lms/djangoapps/courseware/views.py

```diff
@@ -685,6 +685,10 @@ def course_info(request, course_id):
         staff_access = has_access(request.user, 'staff', course)
         masquerade, user = setup_masquerade(request, course_key, staff_access, reset_masquerade_data=True)
 
+        show_enroll_banner = request.user.is_authenticated() and not CourseEnrollment.is_enrolled(user, course.id)
+        if show_enroll_banner and hasattr(course_key, 'ccx'):
```

Comment:
Please add few words to explain that even if the `show_enroll_banner` is `True`, if the key is for a CCX, then the user will be redirected.